CineElorrietaMVC/src/main/java/Vista/PanelPedidos.java

```diff
@@ -0,0 +1,230 @@
+		
+		DireccionLabel = new JLabel("Direccion:\r\n");
+		DireccionLabel.setBounds(221, 60, 66, 14);
+		DireccionLabel.setVisible(false);
+		add(DireccionLabel);
+		
+		Aviso = new JLabel("Rellene los campos");
+		Aviso.setBounds(374, 60, 115, 14);
+		Aviso.setVisible(false);
+		add(Aviso);
+		
+		DireccionTexto = new JTextField();
+		DireccionTexto.setBounds(284, 57, 86, 20);
+		DireccionTexto.setVisible(false);
+		add(DireccionTexto);
+		DireccionTexto.setColumns(10);
+		String variableDeTexto;
+		variableDeTexto =DireccionTexto.getText();
+		
+		String col[] = {"Producto", "Cantidad", "Precio/Cant"};
+		DefaultTableModel tableModel = new DefaultTableModel(col, 0);
+		table = new JTable(tableModel);
+		table.setEnabled(false);
+		table.setModel(new DefaultTableModel(
+			new Object[][] {
+				
+			},
+			new String[] {
+				"Producto", "Cantidad", "Precio/Cant"
+			}
+		));
+		
+		table.setBounds(283, 86, 165, 140);
+		add(table);
+		
+		PrecioFinal = new JLabel("0");
+		PrecioFinal.setBackground(Color.WHITE);
+		PrecioFinal.setForeground(Color.BLACK);
+		PrecioFinal.setBounds(402, 237, 46, 14);
+		add(PrecioFinal);
+		
+		Localidad = new JComboBox();
+		Localidad.addItem("Local");
+		Localidad.addItem("Domicilio");
+		Localidad.setBackground(SystemColor.activeCaption);
+		Localidad.setToolTipText("");
+		Localidad.setBounds(104, 55, 86, 22);
+		add(Localidad);
+		
+		JLabel totalTexto = new JLabel("Total:");
+		totalTexto.setBounds(358, 237, 54, 14);
+		add(totalTexto);
+		
+		btnGuardar = new JButton("Guardar");
+		btnGuardar.setBounds(374, 326, 89, 23);
+		add(btnGuardar);
+		
+		
+		Localidad.addActionListener(new ActionListener() {
+			   @Override
+			   public void actionPerformed(ActionEvent e) {
+				  String Direccion= variableDeTexto;
+				  if (Localidad.getSelectedItem() == "Local") {
+					  DireccionTexto.setVisible(false);
+					  DireccionLabel.setVisible(false);
+					}else if (Localidad.getSelectedItem() == "Domicilio") {
+						DireccionTexto.setVisible(true);
+						DireccionLabel.setVisible(true);
+					}
+			   }
+			});
+		
+		
+		
+		initializeEvents();
+	}
+	
+	private void initializeEvents() {
+		this.btnVolver.addActionListener(listenerBotonVolver(this.controladorPanelPedidos));
+		this.btnAnadir.addActionListener(listenerBotonAnadir(this.controladorPanelPedidos));
+		this.btnGuardar.addActionListener(listenerBotonGuardar(this.controladorPanelPedidos));
+	}
+	
+	private ActionListener listenerBotonVolver(ControladorPanelPedidos controladorPanelPedidos) {
+		return new ActionListener() {
+			public void actionPerformed(ActionEvent arg0) {
+				System.out.println("Ejecutando evento Boton Volver");
+				controladorPanelPedidos.accionadoBottonVolverPanelPedidos();
+			}
+		};
+	}
+	private ActionListener listenerBotonAnadir(ControladorPanelPedidos controladorPanelPedidos) {
+		return new ActionListener() {
+			public void actionPerformed(ActionEvent arg0) {
+				System.out.println("Ejecutando evento Boton A�adir");
```

Comment:
Toda esta lógica de añadir el producto y calcular el total debería hacerse en el modelo de tal forma que desde aquí llamar al controlador con el producto seleccionado y el controlador debería encargarse de añadirlo en producto en el array de productosElegidos del objeto pedido que creáis en el modelo cada vez que abrís la ventana, así en la clase pedido podéis tener dos métodos que os devuelvan por un lado el precio * cantidad del producto y por otro lado el precio total del producto (recorriendo el array de productos elegidos) , de esta forma no tenéis que hacer toda esta logica en la vista

Reply:
Solucionado